pkg/recipes/engine/engine.go

```diff
@@ -226,24 +226,30 @@ func (e *engine) getDriver(ctx context.Context, recipeMetadata recipes.ResourceM
 	return definition, driver, nil
 }
 
-func (e *engine) getRecipeConfigSecrets(ctx context.Context, driver recipedriver.Driver, configuration *recipes.Configuration, definition *recipes.EnvironmentDefinition) (v20231001preview.SecretStoresClientListSecretsResponse, error) {
-	secrets := v20231001preview.SecretStoresClientListSecretsResponse{}
+func (e *engine) getRecipeConfigSecrets(ctx context.Context, driver recipedriver.Driver, configuration *recipes.Configuration, definition *recipes.EnvironmentDefinition) (secretData map[string]map[string]string, err error) {
 	driverWithSecrets, ok := driver.(recipedriver.DriverWithSecrets)
 	if !ok {
-		return secrets, nil
+		return nil, nil
 	}
 
-	secretStore, err := driverWithSecrets.FindSecretIDs(ctx, *configuration, *definition)
+	secretStoreIDResourceKeys, err := driverWithSecrets.FindSecretIDs(ctx, *configuration, *definition)
 	if err != nil {
-		return v20231001preview.SecretStoresClientListSecretsResponse{}, err
+		return nil, err
 	}
 
-	// Retrieves the secret values from the secret store ID provided.
-	if secretStore != "" {
-		secrets, err = e.options.SecretsLoader.LoadSecrets(ctx, secretStore)
+	// Retrieves the secret values from the secret store IDs and keys provided.
+	if secretStoreIDResourceKeys != nil {
+		secretData, err = e.options.SecretsLoader.LoadSecrets(ctx, secretStoreIDResourceKeys)
 		if err != nil {
-			return v20231001preview.SecretStoresClientListSecretsResponse{}, recipes.NewRecipeError(recipes.LoadSecretsFailed, fmt.Sprintf("failed to fetch secrets from the secret store resource id %s for Terraform recipe %s deployment: %s", secretStore, definition.TemplatePath, err.Error()), util.RecipeSetupError, recipes.GetErrorDetails(err))
+			// Get list of secretStoreIDs to log in the error message.
+			secretStoreIDKeys := []string{}
+			for secretStoreIDKey := range secretStoreIDResourceKeys {
+				secretStoreIDKeys = append(secretStoreIDKeys, secretStoreIDKey)
+			}
+
+			return nil, recipes.NewRecipeError(recipes.LoadSecretsFailed, fmt.Sprintf("failed to fetch secrets from the secret store resource id %s for Terraform recipe %s deployment: %s", strings.Join(secretStoreIDKeys, ", "), definition.TemplatePath, err.Error()), util.RecipeSetupError, recipes.GetErrorDetails(err))
```

Comment:
unit tests for this one would be helpful, in the error message i see we are listing all the secret stores ids. It maybe the case that LoadSecrets failed for only one secret store

Reply:
added test here: https://github.com/radius-project/radius/pull/7695/files?diff=unified&w=0#diff-3fdb9931c1f2ddfc5b14d87f47b4124f2bdc1da187219fa1ad9cbe79fc326528L319